Bug: serialization issues after torch 2.6

### Description

on jan 29, torch 2.6 introduced some disruptive changes for users serializing more than basic types (https://github.com/pytorch/pytorch/releases/tag/v2.6.0) and, consequently, our pydra tests `src/tests/utils/data_structures/pydra_helpers_test.py` fail. as a workaround, for now, I have fixed torch version to < 2.6. @wilke0818 and I have been exploring the space but haven't found a solution yet.

### Steps to Reproduce

Install torch 2.6 and run unit tests

### Expected Results

Success

### Actual Results

```
================================================== FAILURES ===================================================
_________________________________________________ test_pydra __________________________________________________
[gw2] darwin -- Python 3.10.10 /Users/fabiocat/Library/Caches/pypoetry/virtualenvs/senselab-KP8v1V64-py3.10/bin/python

    def test_pydra() -> None:
        """Test simple tensor pydra workflow."""
        wf = pydra.Workflow(name="wf_test", input_spec=["x"])
        wf.split("x", x=[torch.tensor([[3, 4], [5, 6]]), torch.tensor([[0, 1], [1, 2]])])
    
        wf.add(pydra_task(name="test_task_task", test_input=wf.lzin.x))
>       wf.set_output([("wf_out", wf.test_task_task.lzout.out)])

src/tests/utils/data_structures/pydra_helpers_test.py:19: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = LazyOut(_task=<pydra.engine.task.FunctionTask object at 0x3730e6500>), name = 'out'

    def __getattr__(self, name):
        if name in ("_task", "_attr_type", "_field_names"):
            raise AttributeError(f"{name} hasn't been set yet")
        if name not in self._field_names:
>           raise AttributeError(
                f"Task '{self._task.name}' has no {self._attr_type} attribute '{name}', "
                "available: '" + "', '".join(self._field_names) + "'"
            )
E           AttributeError: Task 'test_task_task' has no output attribute 'out', available: '_is_param', 'all_'

../../Library/Caches/pypoetry/virtualenvs/senselab-KP8v1V64-py3.10/lib/python3.10/site-packages/pydra/engine/specs.py:696: AttributeError
```

### Additional Notes

_No response_

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -29,7 +29,7 @@ packages = [{include = "senselab", from = "src"}]
 [tool.poetry.dependencies]
 python = ">=3.10,<3.13"
 datasets = "~=3"
-torch = ">=2.5,<2.6"
+torch = "~=2.6"
 torchvision = "~=0.20"
 torchaudio = "~=2.5"
 transformers = "~=4.47"

--- a/src/tests/utils/data_structures/pydra_helpers_test.py
+++ b/src/tests/utils/data_structures/pydra_helpers_test.py
@@ -5,6 +5,7 @@ import torch
 
 
 @pydra.mark.task
+@pydra.mark.annotate({"return": {"out": torch.Tensor}})
 def pydra_task(test_input: torch.Tensor) -> torch.Tensor:
     """Task function for Pydra workflow to run."""
     return test_input + 2
@@ -22,6 +23,9 @@ def test_pydra() -> None:
         sub(wf)
 
     results = wf.result()
-    print("results", results)
-    assert results[0].output.wf_out.equal(torch.tensor([[5, 6], [7, 8]]))
-    assert results[1].output.wf_out.equal(torch.tensor([[2, 3], [3, 4]]))
+    assert results[0].output.wf_out.equal(
+        torch.tensor([[5, 6], [7, 8]])
+    ), f"Expected Tensor of [[5,6],[7,8]] but got {results[0].output.wf_out}"
+    assert results[1].output.wf_out.equal(
+        torch.tensor([[2, 3], [3, 4]])
+    ), f"Expected Tensor of [[2,3],[3,4]] but got {results[1].output.wf_out}"

